third_party/dummy-hook/CMakeLists.txt

```diff
@@ -1,18 +1,27 @@
-if(EXISTS "${Ginkgo_SOURCE_DIR}/.git")
-    set(ADD_HOOK FALSE)
-    set(HOOK_LOCATION "${Ginkgo_SOURCE_DIR}/.git/hooks/pre-commit")
-    if(NOT EXISTS "${HOOK_LOCATION}")
-        set(ADD_HOOK TRUE)
-    else()
-        # check if the correct hook is installed
-        execute_process(COMMAND grep git-cmake-format.py "${HOOK_LOCATION}"
-                        RESULT_VARIABLE res OUTPUT_QUIET)
-        # return value =/= 0 means the pattern was not found
-        if(NOT res EQUAL 0)
+find_package(Git QUIET)
+if(GIT_FOUND)
+    execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --git-path hooks
+      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+      OUTPUT_VARIABLE Ginkgo_GIT_HOOKS_DIR
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    get_filename_component(Ginkgo_GIT_HOOKS_DIR ${Ginkgo_GIT_HOOKS_DIR} REALPATH BASE_DIR ${CMAKE_SOURCE_DIR})
+
+    if(EXISTS "${Ginkgo_GIT_HOOKS_DIR}")
+        set(ADD_HOOK FALSE)
+        set(HOOK_LOCATION "${Ginkgo_GIT_HOOKS_DIR}/pre-commit")
+        if(NOT EXISTS "${HOOK_LOCATION}")
             set(ADD_HOOK TRUE)
+        else()
+            # check if the correct hook is installed
+            execute_process(COMMAND grep git-cmake-format.py "${HOOK_LOCATION}"
+                            RESULT_VARIABLE res OUTPUT_QUIET)
+            # return value =/= 0 means the pattern was not found
+            if(NOT res EQUAL 0)
+                set(ADD_HOOK TRUE)
+            endif()
```

Comment:
this does not work as expect

Reply:
Could you describe the problem? For me, it does the right thing, i.e. adding the dummy hook if pre-commit does not contain `git-cmake-format`.

Reply:
I see. Sorry, I misunderstand this line.
I thought it check git-cmake-format.py is existed or not.
but you check the git-cmake-format.py inside the pre-commit file